src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -4273,3 +4273,34 @@ void C2_MacroAssembler::vector_mask_operation(int opc, Register dst, XMMRegister
   vector_mask_operation_helper(opc, dst, tmp, masklen);
 }
 #endif
+
+void C2_MacroAssembler::vector_maskall_operation(KRegister dst, Register src, int mask_len) {
+  if (VM_Version::supports_avx512bw()) {
+    if (mask_len > 32) {
+      kmovql(dst, src);
+      if (mask_len != 64) {
+        kshiftrql(dst, dst, 64 - mask_len);
+      }
```

Comment:
We are only supporting vector lengths of power of 2 for x86 (8,16,32,64) for byte vector.  The only case that comes here is mask_len == 64.